service/internal/integrationtest/s3/presign_test.go

```diff
@@ -5,49 +5,128 @@ package s3
 import (
 	"bytes"
 	"context"
+	"fmt"
+	"io"
 	"io/ioutil"
 	"net/http"
 	"strconv"
 	"testing"
 	"time"
 
-	"github.com/aws/aws-sdk-go-v2/aws"
+	v4 "github.com/aws/aws-sdk-go-v2/aws/signer/v4"
 	"github.com/aws/aws-sdk-go-v2/service/internal/integrationtest"
 	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/google/go-cmp/cmp"
 )
 
-func TestInteg_PresignURL_PutObject(t *testing.T) {
-	key := integrationtest.UniqueID()
-
-	ctx, cancelFn := context.WithTimeout(context.Background(), 5*time.Second)
-	defer cancelFn()
-
-	cfg, err := integrationtest.LoadConfigWithDefaultRegion("us-west-2")
-	if err != nil {
-		t.Fatalf("failed to load config, %v", err)
+func TestInteg_PresignURL(t *testing.T) {
+	cases := map[string]struct {
+		body                 io.Reader
+		expires              time.Duration
+		sha256Header         string
+		expectedSignedHeader http.Header
+	}{
+		"standard": {
+			body: bytes.NewReader([]byte("Hello-world")),
+			expectedSignedHeader: http.Header{
+				"content-type":   {"application/octet-stream"},
+				"content-length": {"11"},
+			},
+		},
+		"nil-body": {
+			expectedSignedHeader: http.Header{},
+		},
+		// TODO: fix PutObject with an empty body returning 501
+		// "empty-body": {
+		// 	body: bytes.NewReader([]byte("")),
+		// 	expectedSignedHeader: http.Header{
+		// 		"content-type": {"application/octet-stream"},
+		// 	},
+		// },
```

Comment:
Probably don't want to generate PutObject presign url support until this item is fixed.   PutObject with nil body is the common way of using presign PutObject

Reply:
Does nil `body` have the same issue as empty body?